subsys/net/ip/tcp2.c

```diff
@@ -789,8 +808,8 @@ static struct tcp *tcp_conn_new(struct net_pkt *pkt)
 
 	net_context_set_family(conn->context, pkt->family);
 
-	conn->dst = tcp_endpoint_new(pkt, TCP_EP_SRC);
-	conn->src = tcp_endpoint_new(pkt, TCP_EP_DST);
+	conn->dst = tcp_endpoint_set(tcp_endpoint_new(pkt), pkt, TCP_EP_SRC);
```

Comment:
Per your judgement, consider the first comment (tcp_enpoint_set() -> bool), might make the error/reporting handling more clear.